### DISCUSSION TITLE
Encountering unhandled_canonical_code_1 in Google Cloud Storage SDK during GetObjectMetadata operation

### DISCUSSION THREAD

**Discussion by LukePickett2** [googleapis/google-cloud-cpp#14832] [Q&A]
I'm encountering an unhandled_canonical_code_1 error while performing a GetObjectMetadata operation in the Google Cloud Storage SDK for C++. The error appears on version v0.20.x. I’ve looked for documentation regarding this error, but I was unable to find any, either in the SDK or via a google search. This error appears intermittently. It appears that rerunning the same query will avoid this error a subsequent time. Can someone explain what this error means or point me to documentation describing it?

The specific error message is: 
`google-cloud-cpp/google/cloud/storage/internal/logging_client.cc:51 GetObjectMetadata() >> status={Permanent error in GetObjectMetadata: {"error":"unhandled_canonical_code_1"} [INVALID_ARGUMENT] [INVALID_ARGUMENT]}`

**Comment by dbolduc** [comment]
> The error appears on version v0.20.x

We no longer support this branch of `google-cloud-cpp`. We can give you better support on the `v2.x` series.

https://github.com/googleapis/google-cloud-cpp/discussions/11778

https://opensource.google/documentation/policies/library-breaking-change

> `unhandled_canonical_code_1`

Digging around internally, this error seems to come from the OAuth token refresh being cancelled. I do not know how to remedy this error. I suggest updating your version of `google-cloud-cpp` as many bugs have been fixed over time.
